### ISSUE TITLE
Enhancement Request: Introspection/Context-Based Suggestions Starting at Connection Level

### ISSUE THREAD

**Issue by ryan-diekhoff** [dbcodeio/public#157]
Hello Mike - Would it be possible to give users the option of starting introspection at the connection/server level so you can see all of the databases available and then load context-based suggestions for schema and fields once the DB is chosen?  This would come in really handy when building queries that need to join in tables from different databases.  

https://github.com/user-attachments/assets/f09990c1-3654-4c06-a8a7-876dc86f1ecc

**Comment by ryan-diekhoff** [comment]
P.S. Example was based on SQL Server connection in Azure Data Studio.

**Comment by mikeburgh** [comment]
I will need to think about this a little to figure out the solution.

Each database is it's own connection, since most the drivers don't let you change DB's once they are opened, and the schemas are held by the connection. So if the context for autocomplete crosses databases, I will need to track down the other databases connection (or open it if not open) which is doable, they are all grouped by the connection they belong too.

If you have never opened the DB before, the introspection step will be a little slow.. but it gets cached after that first time, and updated in the background on new connections. So if the first time you touch the DB is using the autocomplete it wont have the data, but after that it will be near instant. I will probably start there and see how it goes, I can probably feed some text into the autocomplete suggestions that it's introspecting xxx databases, just so people know what's happening.

**Comment by mikeburgh** [comment]
This is ready to release, it will now suggest, and work with the context for other databases from the connection. Does not handle linked servers yet, as I don't get those from introspection.. not sure that's a good idea anyway to introspect a linked server.

**Comment by mikeburgh** [comment]
Released in version 1.8.0, I will leave this open for a week or so if you notice any issues with it (still other autocomplete issues, but hopefully one less now!)

**Comment by ryan-diekhoff** [comment]
Hi Mike - Thanks for getting this in!  It looks like the introspection works great when starting with database and navigating to the desired table to include in a query.  

I am having some issues pulling up field names in the autocomplete once the table is selected.  I've included a short video to explain.

https://github.com/user-attachments/assets/910567c1-2fd9-476f-8a1c-6e7fa1ab264c

**Comment by mikeburgh** [comment]
Thanks for the video.. issue is the parser choking on the column for some reason, and not recovering.. works in a where clause.. 

<img width="722" alt="Image" src="https://github.com/user-attachments/assets/b453d81e-21ea-491e-9149-2d799f600d05" />

Digging in to see why the parser does not recover/chokes.

**Comment by mikeburgh** [comment]
Found it, fixed it, will update when released.

**Comment by mikeburgh** [comment]
Released in 1.8.2, going to close this, but if you find more issues with it.. re open..

**Comment by ryan-diekhoff** [comment]
Thanks!  It looks like the field name suggestions are tied to the database currently connected to the editor instead of database in the FROM clause unless an alias is used.  I've attached a video to show further:

https://github.com/user-attachments/assets/f1c560e7-0756-4234-b479-12179992cc9d

**Comment by mikeburgh** [comment]
Just to confirm.. do you have the same table in both the UTIL_Job and UTIL_Archive ? 

I don't have overlapping tables, and it works (see below).... . but given the names (Job/Archive) I am guessing that's it.. got to think a little about how to resolve that and get it to pick the right one.


https://github.com/user-attachments/assets/d05fc9e9-585f-4336-8c38-9ada0c94e8a1

**Comment by ryan-diekhoff** [comment]
No, the table name in video doesn't overlap between the two db's.  Are you saying it could be the underscores in the db names that could be causing some issues or something else?

I will try the scenario again with some other databases to see if I get the same results.  Thanks.

**Comment by mikeburgh** [comment]
Not sure yet what it is, just wanted to confirm the table Rule_History was not in both databases.. I will do some more digging to see if I can replicate it on my end.

**Comment by mikeburgh** [comment]
Tracked it down and fixed, will be in the next release

**Comment by mikeburgh** [comment]
Should be fixed now in 1.10.0  re open if you can find other issues with this.
